src/backend/distributed/commands/cascade_table_operation_for_connected_relations.c

```diff
@@ -623,16 +623,20 @@ ExecuteForeignKeyCreateCommand(const char *commandString, bool skip_validation)
 	 */
 	Assert(IsA(parseTree, AlterTableStmt));
 
+	bool oldValue = SkipConstraintValidation;
+
 	if (skip_validation && IsA(parseTree, AlterTableStmt))
 	{
-		parseTree =
-			SkipForeignKeyValidationIfConstraintIsFkey((AlterTableStmt *) parseTree,
-													   true);
+		set_config_option("citus.skip_constraint_validation", "true",
```

Comment:
given the messy API, it would be nice to create a wrapper for this call like we've done for some other GUCs